src/test/integration/users/AVS.t.sol

```diff
@@ -0,0 +1,199 @@
+
+        allocationManager.slashOperator(p);
+    }
+
+    function slashOperator(User operator, uint32 operatorSetId) public {
+        slashOperator(operator, operatorSetId, cheats.randomUint(0.1 ether, 1 ether));
+    }
+
+    function deregisterFromOperatorSets(User operator, uint32[] memory operatorSetIds) public createSnapshot {
+        _logM("deregisterFromOperatorSets");
+
+        DeregisterParams memory p = DeregisterParams({
+            operator: address(operator),
+            avs: address(this),
+            operatorSetIds: operatorSetIds
+        });
+
+        _logDeregisterFromOperatorSets(p);
+
+        allocationManager.deregisterFromOperatorSets(p);
+    }
+
+    function setAVSRegistrar(
+        IAVSRegistrar registrar
+    ) public createSnapshot {
+        _logM("setAVSRegistrar");
+
+        console.log("Setting AVS registrar to: %s", address(registrar));
+
+        allocationManager.setAVSRegistrar(registrar);
+    }
+
+    function addStrategiesToOperatorSet(uint32 operatorSetId, IStrategy[] memory strategies) public createSnapshot {
+        _logM("addStrategiesToOperatorSet");
+
+        console.log("Adding strategies to operator set: %d", operatorSetId);
+
+        for (uint256 i; i < strategies.length; ++i) {
+            console.log("   strategy: %s", address(strategies[i]));
+        }
+
+        allocationManager.addStrategiesToOperatorSet(operatorSetId, strategies);
+    }
+
+    function removeStrategiesFromOperatorSet(
+        uint32 operatorSetId,
+        IStrategy[] memory strategies
+    ) public createSnapshot {
+        _logM("removeStrategiesFromOperatorSet");
+
+        console.log("Removing strategies from operator set: %d", operatorSetId);
+
+        for (uint256 i; i < strategies.length; ++i) {
+            console.log("   strategy: %s", address(strategies[i]));
+        }
+
+        allocationManager.removeStrategiesFromOperatorSet(operatorSetId, strategies);
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Logging
+    /// -----------------------------------------------------------------------
+    
+    function _logSlashOperator(
+        SlashingParams memory p
+    ) private pure {
+        console.log("Slashing operator: %s", address(p.operator));
+        console.log("   operatorSetId: %d", p.operatorSetId);
+        console.log("   wadToSlash: %d", p.wadToSlash);
+        console.log("   description: %s", p.description);
+    }
+
+    function _logCreateOperatorSets(
+        CreateSetParams[] memory p
+    ) private pure {
+        console.log("Creating operator sets:");
+        for (uint256 i; i < p.length; ++i) {
+            console.log("   Creating operator set: %d", p[i].operatorSetId);
+            for (uint256 j; j < p[i].strategies.length; ++j) {
+                console.log("       strategy: %s", address(p[i].strategies[j]));
+            }
+        }
+    }
+
+    function _logDeregisterFromOperatorSets(
```

Comment:
Why not have this logging in the logger? Noticed the user contract doesn't have logging built in but uses the logger

Reply:
fixed in [4574f6e](https://github.com/Layr-Labs/eigenlayer-contracts/pull/894/commits/4574f6e615518ffc784794c3be9e53ddbf2f90e6)